File: template/springboot/src/main/java/com/TinyPro/entity/dto/CreatePermissionDto.java
package com.TinyPro.entity.dto;

import jakarta.validation.constraints.NotEmpty;
import lombok.Data;

@Data
public class CreatePermissionDto {
    
    @NotEmpty(message = "{NOT_EMPTY}")
    private String name;
    
    @NotEmpty(message = "{NOT_EMPTY}")
    private String desc;
}

File: template/nestJs/src/permission/dto/create-permission.dto.ts
import { IsNotEmpty } from 'class-validator';
import { i18nValidationMessage } from 'nestjs-i18n';
import { I18nTranslations } from '../../.generate/i18n.generated';
import { ApiProperty } from '@nestjs/swagger';

export class CreatePermissionDto {
  @ApiProperty({
    description: '权限在使用时候的键值对.'
  })
  @IsNotEmpty({
    message: i18nValidationMessage<I18nTranslations>('validation.NOT_EMPTY'),
  })
  name: string;

  @ApiProperty({
    description: '权限在后台, 人类可读的短简介'
  })
  @IsNotEmpty({
    message: i18nValidationMessage<I18nTranslations>('validation.NOT_EMPTY'),
  })
  desc: string;
}
